### PR TITLE
TMEDIA-634: Add in SASS integrations.

### PR DESCRIPTION
**Pull request by nschubach**

### CODE REVIEW

File: css/index.css [hunk 1]
```diff
@@ -0,0 +1,70 @@
+/* http://meyerweb.com/eric/tools/css/reset/
```
Comment (matthewroach) [inline css/index.css]: This is a compiled css file did you mean to commit this?

File: css/index.css.map [hunk 2]
```diff
@@ -0,0 +1 @@
+{"version":3,"sourceRoot":"","sources":["../src/scss/root/_reset.scss","../src/scss/root/_a11y.scss","../src/scss/root/_index.scss"],"names":[],"mappings":"AAAA;AAAA;AAAA;AAAA;AAKA;AAAA;AAAA;AAAA;AAAA;AAAA;AAAA;AAAA;AAAA;AAAA;AAAA;AAAA;AAAA;EAaC;EACA;EACA;EACA;EACA;EACA;;;AAED;AACA;AAAA;EAEC;;;AAED;EACC;;;AAED;EACC;;;AAED;AAAA;EAEC;EACA;;;AAED;EACC;EACA;;;AAGD;EACC;EACA;EACA;;;ACjDD;EACC;EACA;EACA;EACA;EACA;EACA;EACA;;;ACoDD;AAgDA","file":"index.css"}
```
Comment (matthewroach) [inline css/index.css.map]: This goes with the `index.css` file - did you mean to commit it?

File: src/scss/root/_index.scss [hunk 3]
```diff
@@ -0,0 +1,112 @@
+@use 'sass:list';
+@use 'sass:map';
+
+@forward 'reset';
```
Comment (matthewroach) [inline src/scss/root/_index.scss]: Shouldn't this be `@use`? We just want the contents output - consumers don't need access to anything in these files

File: src/scss/root/_index.scss [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+
+@function component-var($component, $item) {
+	@return var(--c-#{$component}-#{$item});
+}
+
+@function block-var($block, $item) {
+	@return var(--b-#{$block}-#{$item});
+}
+
+@mixin set-alias($alias, $items) {
```
Comment (matthewroach) [inline src/scss/root/_index.scss]: Are the `set-X` mixin's needed now with the way `@use` works?

File: package.json [hunk 5]
```diff
@@ -55,13 +58,24 @@
     "eslint-plugin-react": "^7.28.0",
     "eslint-plugin-react-hooks": "^4.3.0",
     "eslint-plugin-sort-exports": "^0.6.0",
-    "hygen": "^6.1.0",
+    "globby": "^13.1.0",
```
Comment (matthewroach) [inline package.json]: What's `globby` used for?

File: package.json [hunk 6]
```diff
@@ -55,13 +58,24 @@
...
     "husky": "^7.0.4",
+    "hygen": "^6.1.0",
     "jest": "^27.4.7",
+    "mini-css-extract-plugin": "^2.5.2",
+    "postcss": "^8.4.5",
```
Comment (matthewroach) [inline package.json]: Are all these packages needed?

**Review by matthewroach — CHANGES_REQUESTED** [review]

File: css/index.css (hunk 1 above)
Comment (nschubach) [reply css/index.css]: Nope, removing

File: css/index.css.map (hunk 2 above)
Comment (nschubach) [reply css/index.css.map]: removing

File: src/scss/root/_index.scss (hunk 4 above)
Comment (nschubach) [reply src/scss/root/_index.scss]: nope, I must have copied those back in...

File: package.json (hunk 5 above)
Comment (nschubach) [reply package.json]: was a test, removing.

File: package.json (hunk 6 above)
Comment (nschubach) [reply package.json]: also... not needed.

File: src/scss/root/_index.scss (hunk 3 above)
Comment (nschubach) [reply src/scss/root/_index.scss]: changed, tested, not needed to be forwarded

File: src/scss/root/_index.scss [hunk 7]
```diff
@@ -0,0 +1,102 @@
...
+
+@function component-var($component, $item) {
+	@return var(--c-#{$component}-#{$item});
+}
+
+@function block-var($block, $item) {
+	@return var(--b-#{$block}-#{$item});
+}
+
+/* stylelint-disable max-nesting-depth, no-invalid-position-at-import-rule */
```
Comment (matthewroach) [inline src/scss/root/_index.scss]: Can we switch all comments to use `//` to avoid additional overhead for Sass compilation and outputting of comments

Comment (nschubach) [reply src/scss/root/_index.scss]: as long as that's not going to get in the way of legal issues... should we include the:
```
/* http://meyerweb.com/eric/tools/css/reset/
   v2.0 | 20110126
   License: none (public domain)
*/
```

Comment (nschubach) [reply src/scss/root/_index.scss]: I suppose since it's public domain, it's free to edit..

**Review by matthewroach — APPROVED** [review]

**Pull request merged**